### PR TITLE
DocSearch v3 migration

### PR DESCRIPTION
**Pull request by fabianrbz**
### Description

Migrate to DocSearch v3 and use the new index config.

### Testing instructions

Preview link: 

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

**Comment by lena-larionova** [comment]
Will also resolve https://github.com/Kong/docs.konghq.com/issues/6447.

**Comment by cloudjumpercat** [comment]
I've got some strange overlapping in the search bar on my screen (laptop) 
Screenshot 2023-11-20 at 3 02 11 PM

**Comment by cloudjumpercat** [comment]
I'm not sure what the limitations of the search are, but wanted to share a few of my findings.

When I search for "jwt token" from the Mesh docs, I get these results. These look fine to me.
Screenshot 2023-11-20 at 3 04 16 PM


But when I search "jwt token mesh" from the docs homepage, I get different results. And I'm not sure why it doesn't highlight the keywords from the search (like "JWT token") like it did in the previous search. It seems like because it's not highlighting the keywords, the returned search result "phrases" don't seem as useful/tell me if what I'm looking for is on that page.
Screenshot 2023-11-20 at 3 04 53 PM

From the Konnect docs I searched for "service mesh", but why does it put results with the words just "service"  or just "mesh" before results that contain both keywords? I would've expected those with "service mesh" to come up earlier in the search results than pages that just have "service" or "mesh".
Screenshot 2023-11-20 at 3 05 49 PM

**Comment by fabianrbz** [comment]
> I've got some strange overlapping in the search bar on my screen (laptop) Screenshot 2023-11-20 at 3 02 11 PM

which browser/resolution?

**Comment by fabianrbz** [comment]
Thanks for the feedback @cloudjumpercat, this is extremely useful!

> I'm not sure what the limitations of the search are, but wanted to share a few of my findings.
> 
> When I search for "jwt token" from the Mesh docs, I get these results. These look fine to me. Screenshot 2023-11-20 at 3 04 16 PM
> 
> But when I search "jwt token mesh" from the docs homepage, I get different results. And I'm not sure why it doesn't highlight the keywords from the search (like "JWT token") like it did in the previous search. It seems like because it's not highlighting the keywords, the returned search result "phrases" don't seem as useful/tell me if what I'm looking for is on that page. Screenshot 2023-11-20 at 3 04 53 PM

I think I fixed this by tweaking the way we [rank](https://www.algolia.com/doc/api-reference/api-parameters/ranking/#ranking-criteria) things. However, I don't know why it didn't highlight the words in the results, that is weird and I will look into it. I know that it is possible for the results not to have highlighted words because we just show an excerpt and it might be that the words are present in the part of the text that we are not showing (This doesn't look like to be 
 what happened in your case).

> From the Konnect docs I searched for "service mesh", but why does it put results with the words just "service" or just "mesh" before results that contain both keywords? I would've expected those with "service mesh" to come up earlier in the search results than pages that just have "service" or "mesh". Screenshot 2023-11-20 at 3 05 49 PM

I made some changes to the index's config, now it will rank higher records with a [higher proximity](https://www.algolia.com/doc/api-reference/api-parameters/ranking/#ranking-criteria). 

Can you take another look, please?

**Comment by lena-larionova** [comment]
@fabianrbz I'm getting this CSS issue as well:

Screenshot 2023-11-21 at 11 26 41 AM

I'm on Chrome, 1470x956px.

File: docs/search.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+   }
+```
+
+
+### Indexing code snippets
+
+By default, Algolia doesn't index code snippets. However, there are a few cases in which we would like to index them.
+For example, code snippets in Troubleshooting sections or used to highlight error messages/codes, etc. Users should be able to copy and paste errors in the search bar and find meaningful results.
+
+To achieve this, we need to add a specific CSS class to the code snippets we want to index (`.algolia-index-code-snippet`) so we can tell the Crawler to extract them.
```
Comment (fabianrbz) [inline docs/search.md:259]: @Kong/team-docs what do you think about this?

**Review by mheap — APPROVED** [review]

**Pull request merged**